### PR TITLE
Use `strict_types` where appropriate

### LINKED ISSUE #9817: Use `strict_types` where appropriate
## Feature Description

Use strict types in PHP files that could reasonably support it. 
```
declare(strict_types=1);
```

This would have to be classes we have full control of the typing and not one using filters.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9817

### CODE REVIEW

File: includes/Analytics.php [hunk 1]
```diff
@@ -71,7 +73,7 @@ public function get_tracking_id(): string {
 		 *
 		 * @var string $tracking_id
 		 */
-		$tracking_id = $this->settings->get_setting( $this->settings::SETTING_NAME_TRACKING_ID );
+		$tracking_id = (string) $this->settings->get_setting( $this->settings::SETTING_NAME_TRACKING_ID );
```
Comment (swissspidy) [inline includes/Analytics.php]: Curious to hear why this was added.

It shouldn't be needed because it ought to be a string already.

Comment (spacedmonkey) [reply includes/Analytics.php]: To the lints pass.

Comment (swissspidy) [reply includes/Analytics.php]: But where did things start to fail now?

Comment (spacedmonkey) [reply includes/Analytics.php]: https://github.com/GoogleForCreators/web-stories-wp/actions/runs/3412221592/jobs/5677445754#step:9:22

Comment (swissspidy) [reply includes/Analytics.php]: That looks more like an issue with PHPStan than anything.

The `@var` block above should mark this as a string and nothing else.

In fact, after rebasing with `main` and removing this cast, there are no PHPStan errors anymore.

So this can be reverted.

Comment (spacedmonkey) [reply includes/Analytics.php]: Reverted. https://github.com/GoogleForCreators/web-stories-wp/pull/12631/commits/80419786946f7a7a52471df74314a0057143f588

**Comment by swissspidy** [comment]
Surprised that there are only 3 test failures.

They're related to https://github.com/GoogleForCreators/web-stories-wp/pull/12631#discussion_r1015660912 but of course not PHPStan errors but PHPUnit test failures.

This is a good reminder to myself that using `register_setting()` with `'type' => 'string'` does not actually cast values or anything. It's just used for the REST API.

To make the cast work & not cause PHPStan complaints, we'd have to do something like this in `get_tracking_id()`:

```php
$tracking_id = $this->settings->get_setting( $this->settings::SETTING_NAME_TRACKING_ID );
return is_scalar( $tracking_id ) ? (string) $tracking_id : '';
```

There are other instances of `$this->settings->get_setting()` where we probably have to do the same, for instance in `Shopify_Query::get_access_token()`.

Needs a closer look & ideally tests.

File: includes/Settings.php [hunk 2]
```diff
@@ -409,15 +409,34 @@ public function get_setting( string $key, $default = false ) {
...
+			 * @var string|array<int|string,mixed>|bool
+			 */
+			$option = get_option( $key );
+		}
+
+		$settings = get_registered_settings();
+		if ( isset( $settings[ $key ] ) ) {
+			$value = rest_sanitize_value_from_schema( $option, $settings[ $key ] );
+			if ( is_wp_error( $value ) ) {
+				return $option;
```
Comment (spacedmonkey) [inline includes/Settings.php:430]: If not valid, should we return the default value?

Comment (swissspidy) [reply includes/Settings.php:430]: Sounds reasonable, just to be on the safe side.

Looks like the function only ever returns errors when using `anyOf` or `uniqueItems`, which we both don't use anyway.

File: includes/Migrations/Update_Publisher_Logos.php [hunk 3]
```diff
@@ -60,7 +60,7 @@ public function __construct( Settings $settings ) {
 	 */
 	public function migrate(): void {
 		$publisher_logo_id       = 0;
-		$publisher_logo_settings = (array) $this->settings->get_setting( $this->settings::SETTING_NAME_PUBLISHER_LOGOS, [] );
+		$publisher_logo_settings = (array) get_option( $this->settings::SETTING_NAME_PUBLISHER_LOGOS, [] );
```
Comment (swissspidy) [inline includes/Migrations/Update_Publisher_Logos.php:63]: Why the change here?

Comment (spacedmonkey) [reply includes/Migrations/Update_Publisher_Logos.php:63]: The old style option does not match the schema,  as it is not an array of integers.

File: includes/Settings.php [hunk 4]
```diff
@@ -318,7 +320,8 @@ public function register(): void {
 				'default'         => [],
 				'show_in_rest'    => [
 					'schema' => [
-						'properties' => [],
+						'properties'           => [],
+						'additionalProperties' => true,
```
Comment (spacedmonkey) [inline includes/Settings.php:324]: This is a fix, this was always wrong.

File: includes/Settings.php [hunk 5]
```diff
@@ -407,15 +410,73 @@ public function get_setting( string $key, $default = false ) {
...
+	}
+
+	/**
+	 * Retrieves all of the registered options for the Settings API.
+	 *
+	 * @since 1.27.0
+	 *
+	 * @return array<string, array<string,string>> Array of registered options.
+	 */
+	protected function get_registered_options(): array {
```
Comment (spacedmonkey) [inline includes/Settings.php:453]: Similar to but not the same as - https://github.com/WordPress/wordpress-develop/blob/302ba4f45f96490f5947f324228f20e0cbe28583/src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L211-L267

Comment (swissspidy) [reply includes/Settings.php:453]: Why do we need a custom version?

What are the exact differences?

Would be good to at least reference the original & the differences in the docblock.

File: includes/Settings.php [hunk 6]
```diff
@@ -407,15 +410,76 @@ public function get_setting( string $key, $default = false ) {
...
+		return $option;
+	}
+
+	/**
+	 * Retrieves all of the registered options for the Settings API.
+	 * Inspired by get_registered_options method found in WordPress. But also get settings that are registered without `show_in_rest` property.
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L211-L267
+	 *
+	 * @since 1.27.0
```
Comment (swissspidy) [inline includes/Settings.php]: We're now at 1.28.0

Comment (spacedmonkey) [inline includes/Settings.php]:
```suggestion
	 * @since 1.28.0
```

File: includes/Settings.php [hunk 7]
```diff
@@ -407,15 +410,76 @@ public function get_setting( string $key, $default = false ) {
...
+		}
+
+		return $option;
+	}
+
+	/**
+	 * Retrieves all of the registered options for the Settings API.
+	 * Inspired by get_registered_options method found in WordPress. But also get settings that are registered without `show_in_rest` property.
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L211-L267
```
Comment (spacedmonkey) [inline includes/Settings.php]:
```suggestion
```

File: includes/Settings.php [hunk 8]
```diff
@@ -407,15 +410,76 @@ public function get_setting( string $key, $default = false ) {
...
+	 *
+	 * @since 1.27.0
+	 *
```
Comment (spacedmonkey) [inline includes/Settings.php:450]:
```suggestion
	 *
	 * @link https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L211-L267
```

File: includes/Settings.php (hunk 5 above)
Comment (spacedmonkey) [reply includes/Settings.php:453]: Added a comment and link. It is not a little different. It is mostly because that method is part of settings rest api, so not easy to reuse the original method.

**Review by swissspidy — APPROVED** [review]
